applications/app/templates/serviceWorker.scala.js

```diff
@@ -8,201 +8,267 @@
-    };
-    // To workaround a bug in Chrome which results in broken HTTPS->HTTP
-    // redirects, we only handle root requests if they match the developer
-    // blog. The info section often hosts holding pages which will could
-    // eventually redirect to a HTTP page.
-    // https://github.com/guardian/frontend/issues/10936
-    var isRequestToDeveloperBlog = url.pathname.match(/^\/info\/developer-blog($|\/.*$)/);
-    if (isRootRequest && isRequestToDeveloperBlog && doesRequestAcceptHtml(request)) {
-        // HTML pages fallback to offline crossword
-        event.respondWith(
-            fetch(request)
-                .catch(function () {
+            event.respondWith(
+                fetch(request).catch(function () {
                     return caches.match('/offline-crossword');
                 })
-        );
-    } else if (isAssetRequest) {
-        // Default fetch behaviour
-        // Cache first for all other requests
-        event.respondWith(
-            caches.match(request)
-                .then(function (response) {
-                    // Workaround Firefox bug which drops cookies
-                    // https://github.com/guardian/frontend/issues/12012
-                    return response || fetch(request, needCredentialsWorkaround(request.url) ? { credentials: 'include' } : {});
-                })
-        );
+            );
+        };
     }
-});
+})();
+
+var requestForAsset = (function (event) {
+    var isAssetRequest = (function (url) {
+        var assetPathRegex = new RegExp('^@Configuration.assets.path');
+        return function (url) {
+            @if(play.Play.isDev()) {
+                return assetPathRegex.test(url.pathname);
+            } else {
+                return assetPathRegex.test(url.href);
+            }
+        }
+    })();
+
+    return function (event) {
+        var request = event.request;
+        var url = new URL(request.url);
+
+        if (isAssetRequest(url)) {
+            // Default fetch behaviour
+            // Cache first for all other requests
+            event.respondWith(
+                caches.match(request)
+                    .then(function (response) {
+                        // Workaround Firefox bug which drops cookies
+                        // https://github.com/guardian/frontend/issues/12012
+                        return response || fetch(request, needCredentialsWorkaround(request.url) ? {
+                            credentials: 'include'
+                        } : {});
+                    })
+            );
+        }
+    }
+})();
+
+/**
+ * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
+ * EVENT LISTENERS
+ */
 
-self.addEventListener('activate', function(event) {
+this.addEventListener('install', function (event) {
+    event.waitUntil(updateOfflineCrosswordCache());
 });
 
-self.addEventListener('push', function(event) {
+self.addEventListener('activate', function (event) {});
```

Comment:
fair enough, didn't want to make too many edits to functionality, but yeah, not doing much by the look of it